src/driver/op.rs

```diff
@@ -83,57 +101,72 @@ impl From<cqueue::Entry> for CqeResult {
     }
 }
 
-impl<T, CqeType> Op<T, CqeType>
-where
-    T: Completable,
-{
+impl<T, CqeType> Op<T, CqeType> {
     /// Create a new operation
-    fn new(data: T, inner: &mut driver::Driver) -> Self {
+    fn new(data: T, index: usize) -> Self {
         Op {
-            index: inner.ops.insert(),
+            index,
             data: Some(data),
             _cqe_type: PhantomData,
             _phantom: PhantomData,
         }
     }
 
-    /// Submit an operation to uring.
+    /// Submit an operation to the driver.
     ///
-    /// `state` is stored during the operation tracking any state submitted to
+    /// `data` is stored during the operation tracking any state for submission to
     /// the kernel.
-    pub(super) fn submit_with<F>(data: T, f: F) -> io::Result<Self>
+    pub(super) fn submit_with<F>(mut data: T, f: F) -> Self
     where
         F: FnOnce(&mut T) -> squeue::Entry,
     {
         CONTEXT.with(|cx| {
             cx.with_driver_mut(|driver| {
-                // Create the operation
-                let mut op = Op::new(data, driver);
+                // Get an vacent entry in the Lifecycle slab
+                let entry = driver.ops.insert();
 
-                // Configure the SQE
-                let sqe = f(op.data.as_mut().unwrap()).user_data(op.index as _);
+                // Configure the Sqe for submission
+                let sqe = f(&mut data).user_data(entry.key() as _);
 
-                // Push the new operation
-                while unsafe { driver.uring.submission().push(&sqe).is_err() } {
-                    // If the submission queue is full, flush it to the kernel
-                    driver.submit()?;
-                }
+                // Create a pending Lifecycle entry for the Op
+                let op = Op::new(data, entry.key());
+                entry.insert(Lifecycle::Pending(sqe));
 
-                Ok(op)
+                op
             })
         })
     }
 
-    /// Try submitting an operation to uring
-    pub(super) fn try_submit_with<F>(data: T, f: F) -> io::Result<Self>
-    where
-        F: FnOnce(&mut T) -> squeue::Entry,
-    {
-        if CONTEXT.with(|cx| cx.is_set()) {
-            Op::submit_with(data, f)
-        } else {
-            Err(io::ErrorKind::Other.into())
-        }
+    /// Enqueue an operation on the submission ring, without `await`ing the future
```

Comment:
Could end the sentence with a period, to be consistent with other changes to this file.